docs/contribute/framework.md

```diff
@@ -0,0 +1,145 @@
+
+- `exports`: The root, `./preset`, and `package.json` exports are required. If your framework has a `preview.js`, then that is required as well.
+- `types`: We strongly encourage you to author your framework in TypeScript and distribute the types.
+- `dependencies` and `devDependencies`: These are just examples. Yours may look quite different.
+- `peerDependencies`: If your framework provides support for multiple versions of the libraries you’re targeting, be sure that is represented here.
+
+</details>
+
+**`preset.js`** ([example](https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/src/preset.ts))
+
+The [preset API](../addons/writing-presets) is where you will configure the Storybook core (which builder and renderer are used by your framework), the builder (via either the [`webpackFinal`](../builders/webpack#extending-storybooks-webpack-config) or [`viteFinal`](../builders/vite#configuration) export), babel (via the `babel` export), any necessary addons, and any available options for your framework.
+
+**`preview.js`** ([example](https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/src/preview.tsx))
+
+The (optional) [preview API](../configure/overview#configure-story-rendering) is where you configure the rendering of stories, such as global decorators or initializing some runtime config needed for your framework to behave as expected. If your framework requires this file, note that you also need to [configure the `previewAnnotations` in `preset.js`](https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/src/preset.ts#L66-L69).
+
+**`types.ts`** ([example](https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/src/types.ts))
+
+If you author your framework in TypeScript (recommended), you should export the type for `StorybookConfig` which reflects the available options of your framework.
+
+### 5. Test your framework
+
+Test it in a fresh project using a Storybook set up as close as possible to your framework. For example, for `@storybook/nextjs`, which uses React and Webpack5, start with a project that uses `@storybook/react` and `@storybook/builder-webpack5`. Follow the installation instructions from your README and ensure everything works as expected. Remember to test the various versions, configs, and options for the libraries you’re supporting.
+
+### 6. Let us know!
+
+Once it's fully tested and released, please let us know about your framework by either announcing it in the `#showcase` channel of the [Storybook Discord](https://discord.gg/storybook) or tweeting it and mentioning `@storybookjs`. It's our hope that well-made community frameworks can eventually move into the Storybook codebase and be considered "official" support.
+
+## Thank you
+
+We deeply appreciate your help in making Storybook work for more developers and the libraries in which they build. If you have any suggestions to improve this document, please use the “Edit” link below.
```

Comment:
@kylegach, if this is going to be added to the documentation, we don't need this. Instead, we could replace this with something linking to the other parts of the documentation of how to contribute, something like:
```md
---
Learn about other ways you can contribute to Storybook:
- [Documentation](./documentation.md) 
- [Examples](./new-snippets.md) 
- [Code](./code.md)
```

If you're ok with this, we can adjust the other contribution pages